moxin-frontend/src/landing/model_files_item.rs

```diff
@@ -69,13 +74,73 @@ live_design! {
         }
     }
 
-    DownloadPendingButton = <ModelCardButton> {
-        draw_bg: { color: #fff, border_color: #x155EEF, border_width: 0.5}
-        text: "Downloading..."
-        enabled: false
+    DownloadPendingButton = <MoxinButton> {
+        width: 25,
+        height: 25,
+        padding: 4,
+        draw_icon: {
+            fn get_color(self) -> vec4 {
+                return #667085;
+            }
+        }
+    }
 
-        draw_text: {
-            color: #x155EEF;
+    DownloadPendingControls = <View> {
+        align: {y: 0.5},
+        spacing: 8,
+        progress_bar = <View> {
+            width: 74,
+            height: 12,
+            flow: Overlay,
+
+            <RoundedView> {
+                height: Fill,
+                draw_bg: {
+                    color: #D9D9D9,
+                    radius: 2.5,
+                }
+            }
+
+            progress_fill = <RoundedView> {
+                width: 0,
+                height: Fill,
+                draw_bg: {
+                    radius: 2.5,
+                }
+            }
+        }
+        progress_text_layout = <View> {
```

Comment:
Could not align the text properly to the right with empty pad in the left without using a View.